YelpArchive/clusterman
# Auto-detect srv-configs changes with file watchers

By adding some extra functionality/fixing bugs around yelp_batch (v10.1.2), this code is responsible for both adding/creating new watcher files and reloading them periodically. The system is set up to reload PoolManager and Autoscaler objects only if the watcher file has changed to reduce extra work. 

Testing Done:
* Added units to ensure re-instantiation is executed when files are updated.
* `make test`
* Manually tested on kubernetes_autoscaler, drainer, and cluster_metrics instances via `paasta local run`. Test on kubestage by:
    1. Ensure drainer was turned off on kubestage
    2. Running `clusterman manage` and decreasing capacity
    3. See on drainer logs that it was not draining/deleting nodes
    4. Push srv-config change (turning on drainer)
    5. Noticed drainer was draining nodes automatically without restarting service instance.
